pkgs/development/python-modules/python-ironicclient/default.nix

```diff
@@ -0,0 +1,55 @@
+{ lib, buildPythonApplication, fetchPypi
+, pbr, appdirs, cliff, dogpile_cache, jsonschema, keystoneauth1, openstacksdk, osc-lib, oslo-utils, pyyaml, requests, stevedore
+, stestr, requests-mock, oslotest
+}:
+
+buildPythonApplication rec {
+  pname = "python-ironicclient";
+  version = "4.8.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "b55516a72b995f92fb434619cbc1e2effa604c7fcaa6ac4afb8f5af1ea8193a4";
+  };
+
+  buildNativeInputs = [
+    pbr
+  ];
+
+  propagatedBuildInputs = [
+    appdirs
+    cliff
+    dogpile_cache
+    jsonschema
+    keystoneauth1
+    openstacksdk
+    osc-lib
+    oslo-utils
+    pyyaml
+    requests
+    stevedore
+  ];
+
+  checkInputs = [
+    stestr
+    requests-mock
+    oslotest
+  ];
+
+  checkPhase = ''
+    runHook preCheck
```

Comment:
We don't need to add the preHooks because we can't use any of the default phases/hooks anyway.